packages/backend/src/asset/service.ts

```diff
@@ -197,3 +221,13 @@ async function getAssetsPage(
 async function getAll(deps: ServiceDependencies): Promise<Asset[]> {
   return await Asset.query(deps.knex).whereNull('deletedAt')
 }
+
+async function setAssetOn(
+  deps: ServiceDependencies,
+  obj: ToSetOn
+): Promise<void | Asset> {
+  if (!obj) return
+  const asset = await getAsset(deps, obj.assetId)
+  if (asset) obj.asset = asset
+  return asset
+}
```

Comment:
This still feels a bit hacky to me. This is returning the asset but I dont see the `setOn` callers actually ever using the returned value. I realize this is to reduce boilerplate everywhere we are setting this but still wondering if there is a better way utilizing the existing `assetService.get(id)`.

Here is one example but it doesnt quite work. The model wallet address expects assetId and asset to be defined because of the non-null assertion in the model.  Still thinking.

```ts
async function getWalletAddressByUrl(
  deps: ServiceDependencies,
  url: string
): Promise<WalletAddress | undefined> {
  let walletAddress = await WalletAddress.query(deps.knex).findOne({ url })
  if (!walletAddress) return
  // await deps.assetService.setOn(walletAddress)
  walletAddress.asset = await deps.assetService.get(walletAddress.assetId)
  return walletAddress || undefined
}
```